### ISSUE TITLE
Not typesafe?

### ISSUE THREAD

**Issue by JVMartin** [unit-finance/unit-node-sdk#106]
If I'm reading the code right, it appears that this SDK is not typesafe in any way, it doesn't use anything like [io-ts](https://gcanti.github.io/io-ts/) or [class-validator](https://github.com/typestack/class-validator) to do runtime type identification on the JSON that is being returned from the server requests, so there's no guarantee the types are safe...

It seems to be just blindly assuming that the JSON returned from the server is shaped the same as the TypeScript types?

In other words, I could think I have the correct shaped object, and then try to reference a property in it, or worse yet a nested property, and that would blow out the stack on my application because it would throw an error trying to reference a property on an object that doesn't exist.

**Comment by JVMartin** [comment]
In other words, I could think I have the correct shaped object, and then try to reference a property in it, or worse yet a nested property, and that would blow out the stack on my application because it would throw an error trying to reference memory that doesn't exist.

**Comment by KATT** [comment]
Presumably, the [JSON Schema](https://docs.unit.co/#json-schema) that this SDK is based on doesn't won't have properties removed without a major API versioning bump, so I wouldn't say that this makes it unsafe & the types can be trusted in the same way one would trust an OpenAPI-Schema.

**Comment by JVMartin** [comment]
Roger that, thanks for the peace of mind! :)
